src/ArmTemplates/Common/API/Clients/Tags/TagClient.cs

```diff
@@ -0,0 +1,69 @@
+﻿// --------------------------------------------------------------------------
+//  <copyright file="TagClient.cs" company="Microsoft">
+//      Copyright (c) Microsoft Corporation. All rights reserved.
+//  </copyright>
+// --------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Abstractions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Tags.Responses;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Constants;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Tags;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Tags
+{
+    public class TagClient : ApiClientBase, ITagClient
+    {
+        const string GetAllTagsLinkedToApiRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/apis/{4}/tags?api-version={5}&format=rawxml";
+        const string GetAllTagsRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/tags?$skip={4}&api-version={5}";
+
+        readonly IApisClient apisClient;
+
+        public TagClient(IApisClient apisClient)
+        {
+            this.apisClient = apisClient;
+        }
+
+        public async Task<string> GetAllAsync(ExtractorParameters extractorParameters, int skipNumOfRecords = 0)
+        {
+            var (azToken, azSubId) = await this.Auth.GetAccessToken();
+
+            var requestUrl = string.Format(GetAllTagsRequest,
```

Comment:
There is a convention to either have a single line with all the params, or every param on a separate line. Although for a string.Format call, this formatting kinda makes sense. I'll leave it up to you if you wanna change it or keep it. (Same thing on line 44/45)

Reply:
I know about that convention and I have the same thoughts as you - for string.Format it is making sense :D
I'll leave it as it is for now, we will maybe go to NuGet package in nearest future and will get rid of REST requests at all.